src/credentials/loginManager.ts

```diff
@@ -21,13 +21,19 @@ export class LoginManager {
     private readonly defaultCredentialsRegion = 'us-east-1'
     private readonly credentialsStore: CredentialsStore = new CredentialsStore()
 
-    public constructor(private readonly awsContext: AwsContext) {}
+    public constructor(
+        private readonly awsContext: AwsContext,
+        public readonly recordAwsSetCredentialsFn: typeof recordAwsSetCredentials = recordAwsSetCredentials
```

Comment:
@hunterwerlla I wonder if we should have these functions defined on a (generated) property of `TelemetryService`, rather than in a bare module. `TelemetryService` is already injected, but then having to inject individual telemetry methods like `recordAwsSetCredentials`, in order to test them, is awkward.

Reply:
You might not need to set up a function injection. [Here](https://github.com/aws/aws-toolkit-vscode/blob/master/src/test/eventSchemas/commands/searchSchemas.test.ts#L107) is an example where a module function is directly stubbed.

Reply:
>  [Here](https://github.com/aws/aws-toolkit-vscode/blob/master/src/test/eventSchemas/commands/searchSchemas.test.ts#L107) is an example where a module function is directly stubbed.

In that example, `vscode.window` is the kind of property I'm proposing we generate. Currently the telemetry functions do not live on such a singleton or injectable.

Reply:
Ah, I understand now. I do like your proposed idea.

Reply:
Why are we testing them though? it should be very rare to have to test telemetry like this directly, ideally we should have integration tests that test telemetry (which is in the backlog)

Reply:
Ok. But it seems arbitrary that we have a TelemetryService but have disconnected it from the `recordXX` functions.